build/patch-charts.sh

```diff
@@ -179,6 +184,15 @@ base:
 \      app: istiod\
 \      istio.io/rev: {{ .Values.revision | default "default" }}' \
     $deployment
+
+  # IOR
+  sed_wrap -i -e '/env:/ a\
+{{- $iorEnabled := "true" }}\
+{{- if and .Values.pilot.routeManager (eq .Values.pilot.routeManager.enabled "false") }}\
+{{- $iorEnabled := "false" }}\
+{{- end }}\
+          - name: ENABLE_IOR\
```

Comment:
i'm not sure what convention upstream uses for things like this, but do we want env, arg, or mesh config (obviously, we can't use mesh config)?  also, for env variable names, should this be prefixed with `ISTIO_META_`?

Reply:
In pilot/istiod, this is called something like feature gate. We enable/disable certain features by making use of env variables. This is the less intrusive approach.